js/src/containers/TwitterEditor.js

```diff
@@ -0,0 +1,98 @@
+/* External dependencies */
+import { compose } from "@wordpress/compose";
+import { withDispatch, withSelect, dispatch as wpDataDispatch } from "@wordpress/data";
+import domReady from "@wordpress/dom-ready";
+
+/* Internal dependencies */
+import TwitterWrapper from "../components/social/TwitterWrapper";
+
+const isPremium = window.wpseoAdminL10n.isPremium;
+
+/**
+ * Container that holds the media object.
+ *
+ * @returns {void}
+ */
+const TwitterContainer = () => {};
+
+TwitterContainer.getMedia = () => {
+	if ( ! TwitterContainer.media ) {
+		TwitterContainer.media = window.wp.media();
+	}
+
+	return TwitterContainer.media;
+};
+
+domReady( () => {
+	const media = TwitterContainer.getMedia();
+	// Listens for the selection of an image. Then gets the right data and dispatches the data to the store.
+	media.on( "select", () => {
+		const selected = media.state().get( "selection" ).first();
+		const {
+			filesizeInBytes,
+			subtype,
+			height,
+			width,
+			url,
+			id,
+			alt,
+		} = selected.attributes;
+		wpDataDispatch( "yoast-seo/editor" ).setTwitterPreviewImage( {
+			bytes: filesizeInBytes,
+			type: subtype,
+			url,
+			id,
+			width,
+			height,
+			alt,
+		} );
+	} );
+	wpDataDispatch( "yoast-seo/editor" ).loadTwitterPreviewData();
+} );
+
+export default compose( [
+	withSelect( select => {
+		const {
+			getTwitterDescription,
+			getTwitterTitle,
+			getTwitterImageUrl,
+			getImageFallback,
+			getTwitterWarnings,
+			getTwitterAlt,
+			getRecommendedReplaceVars,
+			getReplaceVars,
+			getSiteUrl,
+			getAuthorName,
+		} = select( "yoast-seo/editor" );
+		return {
+			imageUrl: getTwitterImageUrl(),
+			imageFallbackUrl: getImageFallback(),
```

Comment:
I am wondering throughout this PR:
Can this not be loaded onto the `imageUrl` via `getTwitterImageUrl()`? This feels a bit top-heavy.

Reply:
Same problem.

We only want to display the fallback in the preview. Not in the actual input fields.